src/core/public/chrome/ui/header/collapsible_nav.tsx

```diff
@@ -0,0 +1,262 @@
+        <EuiCollapsibleNavGroup
+          background="light"
+          className="eui-yScroll"
+          style={{ maxHeight: '40vh' }}
+        >
+          <EuiListGroup
+            aria-label={i18n.translate('core.ui.primaryNav.pinnedLinksAriaLabel', {
+              defaultMessage: 'Pinned links',
+            })}
+            listItems={[
+              {
+                label: 'Home',
+                iconType: 'home',
+                href: homeHref,
+                onClick: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+                  onIsOpenUpdate(false);
+                },
+              },
+            ]}
+            maxWidth="none"
+            color="text"
+            gutterSize="none"
+            size="s"
+          />
+        </EuiCollapsibleNavGroup>
+      </EuiFlexItem>
+
+      <EuiHorizontalRule margin="none" />
+
+      <EuiFlexItem className="eui-yScroll">
+        {/* Recently viewed */}
+        <EuiCollapsibleNavGroup
+          key="recentlyViewed"
+          title={i18n.translate('core.ui.recentlyViewed', { defaultMessage: 'Recently viewed' })}
+          isCollapsible={true}
+          initialIsOpen={getIsCategoryOpen('recentlyViewed')}
+          onToggle={isCategoryOpen => setIsCategoryOpen('recentlyViewed', isCategoryOpen)}
+        >
+          {recentNavLinks.length > 0 ? (
+            <EuiListGroup
+              aria-label={i18n.translate('core.ui.recentlyViewedAriaLabel', {
+                defaultMessage: 'Recently viewed links',
+              })}
+              listItems={recentNavLinks}
+              maxWidth="none"
+              color="subdued"
+              gutterSize="none"
+              size="s"
+            />
+          ) : (
+            <EuiText size="s" color="subdued" style={{ padding: '0 8px 8px' }}>
+              <p>
+                {i18n.translate('core.ui.EmptyRecentlyViewed', {
+                  defaultMessage: 'No recently viewed items',
+                })}
+              </p>
+            </EuiText>
+          )}
+        </EuiCollapsibleNavGroup>
+
+        {/* Kibana, Observability, Security, and Management sections */}
+        {orderedCategories.map(categoryName => {
+          const category = categoryDictionary[categoryName]!;
+          const links = allCategorizedLinks[categoryName].map(
+            ({ label, href, onClick }: NavLink) => ({
+              label,
+              href,
+              'data-test-subj': 'collapsibleNavAppLink',
+              onClick: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+                onIsOpenUpdate(false);
+                onClick(e);
+              },
+            })
+          );
+
+          return (
+            <EuiCollapsibleNavGroup
+              key={category.label}
+              iconType={category.euiIconType}
+              title={category.label}
+              isCollapsible={true}
+              initialIsOpen={getIsCategoryOpen(category.label)}
```

Comment:
1. Even if localStorage keys can have whitespace chars, as we are not assured that the category's label is a single word, we could have keys such as `core.navGroup.Some Label`

2. More importantly: If you switch language and then reload the app, this persistence is going to be lost, as the key in the new language will not match the previous one.

I would really introduce an `id` property to `AppCategory` to avoid using arbitrary fields as technical ids.